roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -65,16 +75,10 @@ def run(self, tmp, task_vars):
 
     def required_images(self, task_vars):
         deployment_type = get_var(task_vars, "deployment_type")
-        # FIXME(juanvallejo): we should handle gracefully with a proper error
-        # message when given an unexpected value for `deployment_type`.
-        image_base_name = self.docker_image_base[deployment_type]
+        image_base_name = self.docker_image_base.get(deployment_type)
```

Comment:
This means that `image_base_name` may be `None`, and then in the `image_from_base_name` method we treat it as a dict and may get a `TypeError` exception.

Since we already check the deployment type in `is_active`, here we can access it directly as it was before.

The handling of unexpected value for `deployment_type` from the FIXME comment has been resolved by skipping the check (though we do not currently produce messages to explain why a check was skipped, potential future feature).